### PR TITLE
Update Python versions run in CI, update tests to use cffconvert 2

### LINKED ISSUE #441: CI tests fail due to unsupported Python version in latest Ubuntu runner
Currently we test against Python 3.6-3.9 in CI on a `ubuntu-latest` runner. However, since GitHub have updated the latest runner to be `ubuntu-22.04` which doesn't provide Python 3.6, the test run fails for this Python version.

To fix this, we could stop testing against 3.6, and instead add the newer Python versions to the matrix.

### PR DESCRIPTION
**Pull request by sdruskat**
- Recent PRs against `main` failed because Python 3.6. is no longer supported in Ubuntu latest, which is 22.04.

**Related issues**

Refs: https://github.com/actions/setup-python/issues/544

Fixes #441 

**Describe the changes made in this pull request**

- Remove Python 3.6 (end of life 2021-12-23) and add newer versions up to 3.11
- Update script for testing examples to use `cffconvert` (latest, i.e., >=2.0)
- Except two links from link checking as they continuously fail, which may be because they're behing a proxy that doesn't allow this kind of access

**Review checklist**

- [ ] Please check if the pull request is against the correct branch  
(format/schema/semantic documentation changes: `develop`; typos, meta files, etc.: `main`)
- [ ] Please check if all changes are recorded in `CHANGELOG.md` and adapt if necessary.
- [ ] Please run tests locally.

```bash
cd $(mktemp -d --tmpdir cff.XXXXXX)
git clone https://github.com/citation-file-format/citation-file-format .
git checkout update-python-versions-for-CI
python3 -m venv env
source env/bin/activate
pip install --upgrade pip wheel setuptools
pip install -r requirements.txt
pytest
<do other things>
```

- [ ] Check if correct Python versions are run and test successfully

### CODE REVIEW

**Review by jspaaks — APPROVED** [review]
Sorry for the long wait!

**Comment by jspaaks** [comment]
Some of the links appear dead but are actually ok when I check them; the `testing/test` jobs require a bit more investigation though, they generate a "The namespace_packages parameter is deprecated." error, seemingly during pip install of a dependency, ruamel.yaml

**Comment by sdruskat** [comment]
> Some of the links appear dead but are actually ok when I check them; 

I've included link checking exceptions for two alive URLs in this PR.

> the `testing/test` jobs require a bit more investigation though, they generate a "The namespace_packages parameter is deprecated." error, seemingly during pip install of a dependency, ruamel.yaml

The error is gone, probably because it has been fixed in a newer release of `ruamel.yaml`.

I'd be very happy for you to do a quick re-review of this @jspaaks. Thanks 🙏.
Skipped CI for the last commit that updated the changelog, so see [the previous commit](https://github.com/citation-file-format/citation-file-format/pull/424/commits/b0cd272a13bb98902ed466ef06ee12da8597edea) for the successfully run CI jobs.

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -14,13 +14,14 @@ jobs:
       fail-fast: false
       matrix:
         os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
-        python-version: ['3.6', '3.7', '3.8', '3.9']
+        python-version: ['3.7', '3.8', '3.9', '3.8', '3.9', '3.10', '3.11']
```
Comment (jspaaks) [inline .github/workflows/testing.yml:17]:
```suggestion
        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
```
probably what you meant :)

Comment (sdruskat) [reply .github/workflows/testing.yml]: Oops, nice catch, thanks.

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-cffconvert==1.*
+cffconvert
```
[the remark this answers is not in the document]
Comment (sdruskat) [reply requirements.txt:1]: I think I've made the necessary changes in `examples/validate.py`? Adapted to the new API anyway ...

[the remark this answers is not in the document]
Comment (jspaaks) [reply requirements.txt:1]: Sorry my bad, not sure what I was looking at. Made #445 anyway.

**Review by jspaaks — APPROVED** [review]
LGTM, thanks for the fixes. Merge at your convenience.

**Pull request merged**